### PR TITLE
Adjust integration test timeouts

### PR DESCRIPTION
**Pull request by jakule**
After running some tests, I discovered that many of our tests fail because of high CPU usage. Our system tests run many tasks in the background, making them flaky and hard to debug. Reducing background tasks (changing frequency from 100ms to about 3s) should help with that and reduce warnings like this:
```
2022-11-17T20:10:15Z WARN [SQLITE]    "SLOW TRANSACTION: 2.003737548s, goroutine 235361 [running]:\nruntime/debug.Stack()\n\t/opt/go/src/runtime/debug/stack.go:24 +0x72\ngithub.com/gravitational/teleport/lib/backend/lite.(*Backend).inTransaction.func1()\n\t/workspace/lib/backend/lite/lite.go:938 +0x85\ngithub.com/gravitational/teleport/lib/backend/lite.(*Backend).inTransaction(0xc000e59340, {0x9f209e8, 0xc0074fc300}, 0xc009fa2358)\n\t/workspace/lib/backend/lite/lite.go:943 +0x2c5\ngithub.com/gravitational/teleport/lib/backend/lite.(*Backend).GetRange(0xc000e59340, {0x9f209e8, 0xc0074fc300}, {0xc0038b07e0, 0xf, 0x10}, {0xc0038b0811, 0xf, 0xf}, 0x0)\n\t/workspace/lib/backend/lite/lite.go:735 +0x373\ngithub.com/gravitational/teleport/lib/backend.(*Sanitizer).GetRange(0xc007f26900, {0x9f209e8, 0xc0074fc300}, {0xc0038b07e0, 0xf, 0x10}, {0xc0038b0811, 0xf, 0xf}, 0x0)\n\t/workspace/lib/backend/sanitize.go:63 +0x11b\ngithub.com/g...
```

This PR also temporarily disables `TestProxyTunnelStrategyProxyPeering()` as this test seems to be very sensitive to any timeout changes.

### CODE REVIEW

**Review by gabrielcorado — APPROVED** [review]

File: integration/helpers/timeouts.go [hunk 1]
```diff
@@ -30,9 +30,10 @@ func SetTestTimeouts(t time.Duration) {
 	// TODO(tcsc): Remove this altogether and replace with per-test timeout
 	//             config (as per #8913)
 
-	apidefaults.SetTestTimeouts(t, t)
+	// Randomize the times a little as we don't want to trigger all tasks at the exact same time.
```
Comment (codingllama) [inline integration/helpers/timeouts.go:33]:
```suggestion
	// Space out the timeouts a little, as we don't want to trigger all tasks at
	// the exact same time.
```

"Randomize" makes me think there's a random number somewhere, although maybe that's just me.

File: integration/helpers/timeouts.go [hunk 2]
```diff
@@ -30,9 +30,10 @@ func SetTestTimeouts(t time.Duration) {
...
+	apidefaults.SetTestTimeouts(time.Duration(float64(t)*1.0), time.Duration(float64(t)*1.1))
 
-	defaults.ResyncInterval = t
-	defaults.SessionRefreshPeriod = t
-	defaults.HeartbeatCheckPeriod = t
+	defaults.ResyncInterval = time.Duration(float64(t) * 0.9)
+	defaults.SessionRefreshPeriod = time.Duration(float64(t) * 1.2)
+	defaults.HeartbeatCheckPeriod = time.Duration(float64(t) * 1.3)
```
Comment (codingllama) [inline integration/helpers/timeouts.go:38]:
```suggestion
	defaults.ResyncInterval = time.Duration(float64(t) * 1.2)
	defaults.SessionRefreshPeriod = time.Duration(float64(t) * 1.3)
	defaults.HeartbeatCheckPeriod = time.Duration(float64(t) * 1.4)
```

so it's never less than the original timeout?

File: integration/helpers/trustedclusters.go [hunk 3]
```diff
@@ -138,6 +138,6 @@ func WaitForActiveTunnelConnections(t *testing.T, tunnel reversetunnel.Server, c
 	},
 		30*time.Second,
 		time.Second,
-		"Active tunnel connections did not reach %v in the expected time frame", expectedCount,
+		"Active tunnel connections did not reach %v in the expected time frame", expectedCount, 30*time.Second,
```
Comment (codingllama) [inline integration/helpers/trustedclusters.go:141]: Missing format for the last param?

File: integration/proxy/proxy_helpers.go [hunk 4]
```diff
@@ -593,12 +593,12 @@ func mustStartMockALBProxy(t *testing.T, proxyAddr string) *mockAWSALBProxy {
...
 // waitForActivePeerProxyConnections waits for remote cluster to report a minimum number of active proxy peer connections
-func waitForActivePeerProxyConnections(t *testing.T, tunnel reversetunnel.Server, expectedCount int) {
+func waitForActivePeerProxyConnections(t *testing.T, tunnel reversetunnel.Server, expectedCount int) { //nolint:unused
 	require.Eventually(t, func() bool {
 		return tunnel.GetProxyPeerClient().GetConnectionsCount() >= expectedCount
 	},
 		30*time.Second,
 		time.Second,
-		"Peer proxy connections did not reach %v in the expected time frame", expectedCount,
+		"Peer proxy connections did not reach %v in the expected time frame", expectedCount, 30*time.Second,
```
Comment (codingllama) [inline integration/proxy/proxy_helpers.go:602]: Ditto.

File: integration/proxy/proxy_helpers.go [hunk 5]
```diff
@@ -593,12 +593,12 @@ func mustStartMockALBProxy(t *testing.T, proxyAddr string) *mockAWSALBProxy {
 }
 
 // waitForActivePeerProxyConnections waits for remote cluster to report a minimum number of active proxy peer connections
-func waitForActivePeerProxyConnections(t *testing.T, tunnel reversetunnel.Server, expectedCount int) {
+func waitForActivePeerProxyConnections(t *testing.T, tunnel reversetunnel.Server, expectedCount int) { //nolint:unused
```
Comment (codingllama) [inline integration/proxy/proxy_helpers.go:596]: Why is the `unused` necessary here?

Could you add a comment after the directive explaining that? Eg: `//nolint:unused // <Explanation here>`. (See example in [Nolint Directive](https://golangci-lint.run/usage/false-positives/#nolint-directive).)

**Review by codingllama — COMMENTED** [review]
Thanks for the fix, Jakub!

File: integration/helpers/timeouts.go (hunk 1 above)
Comment (jakule) [reply integration/helpers/timeouts.go:33]: @codingllama Agree, I used XKCD "randomization" https://imgs.xkcd.com/comics/random_number.png

**Review by codingllama — APPROVED** [review]

**Pull request merged**